0006-B-external-service-invocation.md

```diff
@@ -0,0 +1,158 @@
+
+## Related Items
+
+### Related proposals 
+
+Formalizing the proposal here from [this issue](https://github.com/dapr/dapr/issues/4549).
+
+## Expectations and alternatives
+
+* What is in scope for this proposal?
+Feature enhancement to enable external service invocation
+using the existing service invocation building block allowing service communication using HTTP protocol.
+
+* What is deliberately *not* in scope?
+gRPC invocation as well as additional authentication, to include OAuth2,
+is not within scope of this initial proposal and implementation.
+
+
+* What alternatives have been considered, and why do they not solve the problem?
+1. Expanding the existing HTTP Binding.
+2. Creation of another HTTP Binding explicitly dedicated to external service invocation keeping in mind the current pain points.
+
+Moving forward with the alternative approaches goes against the motivation and goal of this proposal,
+as Dapr users would be missing crucial service invocation features,
+be restricted on numerous avenues,
+and be forced to continue abiding by an awkwardly clunky workaround.
+Additional pros/cons may be found in the [linked issue's discussion](https://github.com/dapr/dapr/issues/4549#issuecomment-1414841151).
+
+* Are there any trade-offs being made? (space for time, for example)
+N/A
+
+* What advantages / disadvantages does this proposal have? 
+This proposal allows service invocation to be enabled for non-Dapr endpoints.
+
+Pros:
+- Extends existing service invocation implementation.
+- Same feel as current user invocation process.
+- Can leverage existing service invocation features like resiliency, security practices, observability.
+- Leveraging a new CRD would keep our CRD setup less cluttered and easier to adjust and add to moving forward.
+- With a new CRD you can add/rm endpoints programmatically via kubectl.
+- Allows for user overrides such as base URL and related request information at invocation time.
+
+Cons:
+- Creation of an additional CRD, thus increasing the duplication of boilerplate code for its setup.
+- Need to know external base URL ahead of time to configure, but that may not always be easy for end users.
+- Would need to change the Dapr Operator to notify on edits for external endpoints.
+
+## Implementation Details
+
+### Design
+
+How will this work, technically?
+
+Allow configuration of pieces needed for external service invocation through creation of new CRD titled `ExternalHTTPEndpoint`.
+It is HTTP specific in it's `Kind`.
+This has benefits in being obvious upfront that it supports only `http`,
+and makes it to where we do not need `spec.allowed.protocols`.
+However, it would have the drawback of needing additional CRDs in the future for supporting other protocols such as `gRPC`.
+The sample `yaml` file snippet below represents the proposed configuration.
+
+```
+apiVersion: dapr.io/v1alpha1
+kind: ExternalHTTPEndpoint
+metadata:
+  name: externalserviceinvocation
+spec:
+  allowed:
+  - name: github
```

Comment:
following from this [comment](https://github.com/dapr/dapr/issues/4549#issuecomment-1152624173) on `name` being used for service discovery, if both internal service invocation and external service invocation define the same `name`, one as the `name` field the other in terms of `app-id`, which will be given priority?

Reply:
adding the URL format in this proposal will be helpful.

Reply:
It's the same URL formats as now